sympy/polys/polytools.py

```diff
@@ -7938,3 +7937,248 @@ def named_poly(n, f, K, name, x, polys):
     else:
         poly = Poly.new(poly, head)
     return poly if polys else poly.as_expr()
+
+
+def _find_minimal_factor_combinations(equation_factors):
+    """ Finds all minimal combinations of factors that cover the system of equations. """
+    if not equation_factors:
+        return [set()]
+    elif len(equation_factors) == 1:
+        return [{f} for f in equation_factors[0]]
+
+    eq1, eqs_rest = equation_factors[0], equation_factors[1:]
+    sets = []
+    for f in eq1:
+        eqs_rest_2 = [eq for eq in eqs_rest if f not in eq]
+        sets.extend(s | {f} for s in _find_minimal_factor_combinations(eqs_rest_2))
+    return sets
+
+
+def _combine_factor_systems_poly(factored_eqs):
+    """ Combine factored equation systems and return minimal factor combinations with their conditions. """
+    factor_dict = defaultdict(list)
+    for eq in factored_eqs:
+        for factor, conditions in eq:
+            factor_dict[factor].extend(conditions)
+
+    eq_factors = [{factor for factor, _ in eq} for eq in factored_eqs]
+
+    all_combinations = _find_minimal_factor_combinations(eq_factors)
+
+    minimal_combinations = []
+    for combo in all_combinations:
+        if not any(combo > other for other in all_combinations):
+            minimal_combinations.append(combo)
+
+    return [[(factor, factor_dict[factor]) for factor in combo]
+            for combo in minimal_combinations]
+
+
+def _process_system_equations(eqs, gens=None, **kwargs):
+    """ Process and convert system equations into polynomial form with proper generator handling. """
+    if not eqs:
+        return None, ([], None)
+
+    exprs = [sympify(eq) for eq in eqs]
+    exprs = [expr.simplify() for expr in exprs]
+
+    if all(expr.is_zero for expr in exprs):
+        return None, ([[]], None)
+
+    exprs = [expr for expr in exprs if not expr.is_zero]
+
+    if not exprs:
+        return None, ([[]], None)
+
+    if any(expr.is_number and not expr.is_zero for expr in exprs):
+        return None, ([], None)
+
+    if not gens:
+        try:
+            polys, opt = parallel_poly_from_expr(exprs, **kwargs)
+            gens = opt.gens
+        except (GeneratorsNeeded, PolificationFailed):
+            if all(expr.free_symbols == set() for expr in exprs):
+                if any(not expr.is_zero for expr in exprs):
+                    return None, ([], None)
+                return None, ([[]], None)
+            raise ValueError("Could not determine generators. Please specify them explicitly.")
+
+    try:
+        polys, _ = parallel_poly_from_expr(exprs, gens, **kwargs)
```

Comment:
It isn't necessary to duplicate all of this. You can just pass `*gens` to `parallel_poly_from_expr`.